pkg/api/query/v1.go

```diff
@@ -1132,13 +1150,16 @@ func (qapi *QueryAPI) series(r *http.Request) (interface{}, []error, *api.ApiErr
 		return nil, nil, &api.ApiError{Typ: api.ErrorBadData, Err: err}, func() {}
 	}
 
-	var matcherSets [][]*labels.Matcher
-	for _, s := range r.Form[MatcherParam] {
-		matchers, err := parser.ParseMetricSelector(s)
-		if err != nil {
-			return nil, nil, &api.ApiError{Typ: api.ErrorBadData, Err: err}, func() {}
-		}
-		matcherSets = append(matcherSets, matchers)
+	tenant, err := tenancy.GetTenantFromHTTP(r, qapi.tenantHeader, qapi.defaultTenant, "")
```

Comment:
Maybe we should combine these tenancy check functions (for all handlers), into one convenient function in the tenancy package?

Reply:
you mean something like this?

```
err = AddTenantToContext(....)
if err != nil {
    return ...
}
```

Reply:
I mean something to combine the operation of extracting tenant from header, and then using it to filter/add matchers. So something like, `tenancy.OverrideMatcherSet()` which would return modified matcherSet, ctx and tenant.

For query endpoints, probably something like `tenancy.OverridePromQL()`. Would be easier to maintain and use this. Wdyt? 🙂